src/project-management/project-management.ts

```diff
@@ -152,16 +157,41 @@ export class ProjectManagement implements FirebaseServiceInterface {
    * Lists summary of all apps in the project
    */
   public listAppMetadata(): Promise<AppMetadata[]> {
-    throw new FirebaseProjectManagementError(
-      'service-unavailable', 'This service is not available');
+    return this.requestHandler.listAppMetadata(this.resourceName)
+      .then((responseData) => this.transformResponseToAppMetadata(responseData));
   }
 
   /**
    * Update display name of the project
    */
-  public setDisplayName(displayName: string): Promise<void> {
-    throw new FirebaseProjectManagementError(
-      'service-unavailable', 'This service is not available');
+  public setDisplayName(newDisplayName: string): Promise<void> {
+    return this.requestHandler.setDisplayName(this.resourceName, newDisplayName);
+  }
+
+  private transformResponseToAppMetadata(responseData: any): AppMetadata[] {
+    this.assertListAppsResponseData(responseData, 'listAppMetadata()');
+
+    if (!responseData.apps) {
+      return [];
+    }
+
+    return responseData.apps.map((appJson: any) => {
+      assertServerResponse(
+          validator.isNonEmptyString(appJson.appId),
+          responseData,
+          `"apps[].appId" field must be present in the listAppMetadata() response data.`);
+      assertServerResponse(
+          validator.isNonEmptyString(appJson.platform),
+          responseData,
+          `"apps[].platform" field must be present in the listAppMetadata() response data.`);
+      return {
+        appId: appJson.appId,
+        displayName: appJson.displayName,
```

Comment:
@nbegley , @hiranya911, This makes `displayName` appearing as `undefined` if `displayName` is not specified for the app. Thus for [unit tests](https://github.com/firebase/firebase-admin-node/pull/546/files#diff-d9f4c407f99decc820eed61f77904d19R529) I hadto add a `displayName: undefined` to the expected result for it to pass.
If there's any good way to not include `displayName` if it's `undefined`, I'm happy to go for it

Reply:
Something like:

```
const value = {
  appId: appJson.appId,
  // other fields except displayName
};
if (appJson.displayName) {
  value.displayName = appJson.displayName;
}
return value;
```

Reply:
I will go for this. Thanks @hiranya911.

Reply:
In that case how about this?

```
const value: {[key: string]: any} = {
  appId: appJson.appId,
  // other fields except displayName
};
if (appJson.displayName) {
  value.displayName = appJson.displayName;
}
return value as AppMetadata;
```

Reply:
Using `any` here removes type safety, so in the event something changes, like a new field being added to AppMetadata, the transpiler wont detect that the field is missing from the AppMetadata being returned here.

eg. We'd want this code to show an error that `field1` is not set, but there is no error:
```
interface MyInterface {
    field1: string;
    field2: number;
}

const value: {[key: string]: any} = {};
value.field2 = 1;

let typed : MyInterface = value as MyInterface;
```

So I'd actually prefer to remove `readonly` from all fields in `AppMetadata` and use Hiranya's original suggestion.

Reply:
@hiranya911 Can we have your opinion on removing `readonly`  in `AppMetadata`?

Reply:
I don't feel strongly about it either way. You'd only be removing `readonly` from the internal implementation of `AppMetadata`. You can continue to keep `readonly` in the typings published in the d.ts file.

Reply:
Okay, let's move forward with that then.

Reply:
Thanks @hiranya911. I will move on with removing `readonly` solution

Reply:
Updated